packages/realm/package.json

```diff
@@ -263,13 +263,16 @@
     }
   },
   "dependencies": {
+    "bson": "^4.7.2",
     "debug": "^4.3.4",
     "node-fetch": "^2.6.9",
     "node-machine-id": "^1.1.12",
     "prebuild-install": "^7.1.1"
   },
+  "optionalDependencies": {
```

Comment:
I don't think this is the intended use of `optionalDependencies` as per [this description](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#optionaldependencies):

> If a dependency can be used, but you would like npm to proceed if it cannot be found or fails to install, then you may put it in the optionalDependencies object.

I don't see a valid case where the installation of the `bson` package would fail, as this happens mostly if it fails to download or build a binary for an unsupported target OS or similar failure during a `postinstall` scripts execution.

Reply:
What you might be looking for is the [`peerDependenciesMeta `](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependenciesmeta) which can mark a peer dependency as optional, but I don't think that's what we want, since the internals of our SDK relies on it being installed.

Reply:
I've been experimenting with this, and have also found that the `peerDependency` is the correct way of going about this.  I was under the assumption that `npm` wouldn't install `bson` on `npm install`, but it does.

I think in it's current form on `main` the package.json is fine.  We should stay on v4 for now, as the react-native tests fail using v5

Reply:
Yeah - a missing peer dependency is automatically installed since NPM v7.